src/footer/script1.js

```diff
@@ -0,0 +1,222 @@
+        us_image_parts.pop();
+        const us_image_cdn = us_image_parts.join('/');
+        return us_image_cdn + '/';
+    }
+}
+
+function isCart() {
+    if (getShoptetDataLayer('pageType') == 'cart') {
+        return true;
+    }
+    return false;
+}
+
+async function getResults() {
+    setMainDiv();
+    const us_cart_items = getCartItemsGUIDS();
+    const result = await getRecommnededProductsFromServer(us_cart_items);
+    if (result != null) {
+        await cacheResults(result);
+        await cacheRequest(us_cart_items, result);
+    }
+    printResults();
+}
+
+function setMainDiv() {
+    const us_main_div = document.querySelectorAll('#upsell-container');
+    if (us_main_div.length == 0) {
+        const us_cart = document.querySelectorAll('.cart-table');
+        if (us_cart.length > 0) {
+            us_cart.forEach(async function (el) {
+                el.insertAdjacentHTML('afterend', '<hr><div id="upsell-container"></div>');
+            });
+        }
+    } else {
+        us_main_div.forEach(function (el) {
+            el.innerHTML = '';
+        });
+    }
+}
+
+async function cacheResults(result) {
+    result.recommendations.forEach(async function (recommendation) {
+        recommendation = await getVariantDetail(recommendation);
+        let recommendationJson = JSON.stringify(recommendation);
+        if (recommendationJson != null) {
+            sessionStorage.setItem('us_' + recommendation.id, recommendationJson.toString());
+        }
+    });
+    return true;
+}
+
+async function getVariantDetail(recommendation) {
+    return await getVariantDetailFromEshop(recommendation);
+}
+
+async function getVariantDetailFromEshop(recommendation)
+{
+    const us_link_response = await goToURL(insertCacheInUrl(recommendation.url));
+    const response = await us_link_response.text();
+    const tempContainer = document.createElement('div');
+    tempContainer.innerHTML = response;
+    const availability = tempContainer.querySelector('span.availability-label');
+    recommendation.availability = '-';
+    if (availability) {
+        recommendation.availability = availability.textContent.trim();
+    }
+    const product_id = tempContainer.querySelector('input[name="productId"]');
+    if (product_id) {
+        recommendation.id = product_id.getAttribute('value');
+    } else {
+        return null;
+    }
+    const price_id = tempContainer.querySelector('input[name="priceId"]');
+    if (price_id) {
+        recommendation.priceId = price_id.getAttribute('value');
+    } else {
+        return null;
+    }
+    recommendation.color = null;
+    if (availability.style.color) {
+        recommendation.color = availability.style.color;
+    }
+    const price = tempContainer.querySelector('span.price-final-holder')
+    if (price) {
+        recommendation.price = price.textContent.trim();
+    }
+    if (recommendation.guid)
+    return recommendation;
+}
+
+async function goToURL(url) {
```

Comment:
Trochu mě mate ten název funkce, představil bych si uvnitř nějaký redirect. Funkce jen fetchuje, ale je to na vašem zvážení.

Reply:
Měním na fetchURL, děkuji